tempesta_fw/sock_clnt.c

```diff
@@ -57,7 +57,15 @@ tfw_sock_cli_keepalive_timer_cb(unsigned long data)
 	TfwConnection *conn = (TfwConnection *)data;
 
 	TFW_DBG("Client timeout end\n");
-	ss_close(conn->sk);
+
+	/* Close socket asynchronously to avoid deadlock on del_timer_sync(). */
+	if (ss_close(conn->sk)) {
```

Comment:
This is timer handler, so it's Ok to rearm the timer. However, since keep alive timer expired, it has sense to set lower timeout...